pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -221,6 +231,24 @@ func setVersion(cluster *v3.Cluster) {
 				}
 			}
 		}
+	} else if cluster.Spec.AmazonElasticContainerServiceConfig != nil {
+		setConfigVersion := func(config *v3.MapStringInterface) {
+			v, found := values.GetValue(*config, "kubernetesVersion")
+			if !found || convert.ToString(v) == "" {
+				if minor := cluster.Status.Version.Minor; len(minor) > 1 && (minor[:2] == "10" || minor[:2] == "11") {
```

Comment:
should check that  `cluster.Status.Version` is not nil before doing this

Reply:
kntpick: i think `strings.HasPrefix(minor, "11") || strings.HasPrefix(minor, "11")` is much more readable than `len(minor) > 1 && (minor[:2] == "10" || minor[:2] == "11")`